cornice/validators/__init__.py

```diff
@@ -33,7 +34,8 @@ def extract_cstruct(request):
     if request.content_type == 'application/x-www-form-urlencoded':
         body = request.POST.mixed()
 
-    elif request.content_type and request.content_type != 'application/json':
+    elif (request.content_type and
+          not re.match('^application/(.*?)json$', str(request.content_type))):
```

Comment:
It's expecting ASCII. This is just there because otherwise we cant [mock it](https://github.com/Cornices/cornice/blob/master/tests/test_util.py#L12). But I'll change it to `six.binary_type`.